framework/application/application.h

```diff
@@ -24,31 +24,38 @@
 #ifndef GFXRECON_APPLICATION_APPLICATION_H
 #define GFXRECON_APPLICATION_APPLICATION_H
 
+#include "application/wsi_context.h"
 #include "decode/file_processor.h"
 #include "decode/window.h"
 #include "util/defines.h"
 
+#include <memory>
 #include <string>
+#include <utility>
 #include <vector>
 
 GFXRECON_BEGIN_NAMESPACE(gfxrecon)
 GFXRECON_BEGIN_NAMESPACE(application)
 
-class Application
+class Application final
```

Comment:
There's no "bad" behavior, per say, but I was breaking the functionality of this class (that was used as base) into a different object so that it could be allocated later.  The `final` is really just there so I can be sure that nothing was inheriting from this class any longer.  I can remove it if you prefer.

Reply:
I think using it to debug the PR development is a great tactic!  I can't think of a good reason for Application to be subclassed at this time, since the WSI part has been factored out.  It just made me wonder if there was something in there that must not be overridden.  I think it's okay to leave it and we'll figure it out later if we really need to start subclassing Application.